arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -921,6 +921,35 @@ uint64_t MaintenanceFeature::shardVersion(std::string const& shname) const {
 
 uint64_t MaintenanceFeature::incShardVersion(std::string const& shname) {
   MUTEX_LOCKER(guard, _versionLock);
+
+  uint64_t v = 0;
+  using namespace std::chrono;
+  using clock = steady_clock;
+  auto timeout = duration<double>(600.0);
+  auto stoppage = clock::now();
+  auto snooze = milliseconds(100);
+
+  // Acquire current version from agency and wait for it to have been dealt
+  // with in local current cache. Any future current version will do, as
+  // the version is incremented by the leader ahead of getting here on the
+  // follower.
+  while (!server().isStopping() && clock::now() < stoppage) {
+    cluster::fetchCurrentVersion(0.1 * timeout)
```

Comment:
this will still make a 60 seconds timeout here. Isn't that too much for a single request?

Reply:
In the grand scheme of things this probably does not matter at all. This is just fetching a single number from the agency. It will usually be quick or fail quickly. In the unlikely case that the agency is totally overloaded or somebody sent a STOP signal or misconfigured a firewall or switch and packets are dropped, it is anyway not advisable to move on here. So we might as well wait. I think 60s timeout are OK.